### PR TITLE
🆕 Glance module

### PR DESCRIPTION
**Pull request by hbernardi-luna**
## 📜 Description

👋 New module to make creation of widget easier. 
📜 You can read https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/feature/glance/lbcglance/README.md that will play the role of the documentation!

## 💡 Motivation and Context

🛞 Do not re-invent the wheel each time.

## 📝 Checklist
* [x] I reviewed the submitted code
* [x] I launched `./gradlew detekt`
* [ ] I filled the [changelog](https://github.com/LunabeeStudio/Lunabee-Compose-Android/blob/develop/CHANGELOG.MD)

#### Android 15

https://github.com/user-attachments/assets/a1f05678-5a6a-4163-98fd-41d04a6a197e

#### Android 9

https://github.com/user-attachments/assets/bd655ded-2720-459b-a6e4-7c2054c713d9

### CODE REVIEW

File: app/src/main/res/drawable/ic_view_flipper_1.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="800dp"
```
Comment (PublisherLunabee) [inline app/src/main/res/drawable/ic_view_flipper_1.xml:2]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/res/drawable/ic_view_flipper_1.xml#L2"></span>Limit vector icons sizes to 200×200 to keep icon drawing fast; see https://developer.android.com/studio/write/vector-asset-studio#when for more</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/drawable/ic_view_flipper_2.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="800dp"
```
Comment (PublisherLunabee) [inline app/src/main/res/drawable/ic_view_flipper_2.xml:2]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/res/drawable/ic_view_flipper_2.xml#L2"></span>Limit vector icons sizes to 200×200 to keep icon drawing fast; see https://developer.android.com/studio/write/vector-asset-studio#when for more</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/drawable/ic_view_flipper_3.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="800dp"
```
Comment (PublisherLunabee) [inline app/src/main/res/drawable/ic_view_flipper_3.xml:2]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/res/drawable/ic_view_flipper_3.xml#L2"></span>Limit vector icons sizes to 200×200 to keep icon drawing fast; see https://developer.android.com/studio/write/vector-asset-studio#when for more</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/drawable/ic_view_flipper_4.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="800dp"
```
Comment (PublisherLunabee) [inline app/src/main/res/drawable/ic_view_flipper_4.xml:2]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/res/drawable/ic_view_flipper_4.xml#L2"></span>Limit vector icons sizes to 200×200 to keep icon drawing fast; see https://developer.android.com/studio/write/vector-asset-studio#when for more</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -50,6 +50,10 @@
     <string name="crop_screen_crop_button">Validate</string>
     <string name="crop_screen_retry_button">Retry</string>
 
+    <!-- GlanceScreen -->
+    <string name="glance_title">Glance</string>
+    <string name="glance_description">Glance actions.</string>
```
Comment (PublisherLunabee) [inline app/src/main/res/values/strings.xml:55]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/res/values/strings.xml#L55"></span>The resource <code>R.string.glance_description</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

File: app/src/main/kotlin/studio/lunabee/compose/demo/glance/GlanceScreen.kt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(all = 16.dp),
+    ) {
+        if (pinWidgetHelper.isPinSupported()) {
+            Button(
+                onClick = {
+                    pinWidgetHelper.pin(GlanceWidgetDemo::class.java, GlanceWidgetDemoReceiver::class.java)
```
Comment (PublisherLunabee) [inline app/src/main/kotlin/studio/lunabee/compose/demo/glance/GlanceScreen.kt:54]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/6c09cd48d00137585ef42b09da4d94f399394e06/app/src/main/kotlin/studio/lunabee/compose/demo/glance/GlanceScreen.kt#L54"></span>Call requires API level 26 (current min is 21): <code>pin</code></td>
    </tr>
  </tbody>
</table>

Comment (hbernardi-luna) [reply app/src/main/kotlin/studio/lunabee/compose/demo/glance/GlanceScreen.kt:54]: Fait un effort en fait

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (hbernardi-luna) [reply app/src/main/res/values/strings.xml:55]: Il a raison

File: lbcglance/src/main/kotlin/studio/lunabee/compose/glance/extensions/DpExt.kt
Comment (rlatapy-luna) [inline lbcglance/src/main/kotlin/studio/lunabee/compose/glance/extensions/DpExt.kt:1]: Glance extensions on `Dp` class are too global imo

File: lbcglance/src/main/kotlin/studio/lunabee/compose/glance/helpers/PinWidgetToHomeScreenHelper.kt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+) {
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun <T : GlanceAppWidget, R : GlanceAppWidgetReceiver> pin(
+        widgetClass: Class<T>,
+        receiverClass: Class<R>,
+        prepareIntent: Intent.(existingAppWidgetIds: IntArray) -> Unit = { },
+    ) {
+        val appWidgetManager = AppWidgetManager.getInstance(context)
+        val provider = ComponentName(context, receiverClass)
+        CoroutineScope(dispatcher).launch {
```
Comment (rlatapy-luna) [inline lbcglance/src/main/kotlin/studio/lunabee/compose/glance/helpers/PinWidgetToHomeScreenHelper.kt:37]: why not make the function suspend or inject the coroutine scope directly?

**Review by rlatapy-luna — CHANGES_REQUESTED** [review]
👍

**Review by rlatapy-luna — APPROVED** [review]

**Pull request merged**